app/models/host_aspect_base.rb

```diff
@@ -0,0 +1,25 @@
+class HostAspectBase < ActiveRecord::Base
```

Comment:
> Might make sense to rename HostAspect, as it's causing confusion.

I am not too attached to this name, suggestions?

> There is no relation corresponding to HostAspect.execution_model here, is that intentionally?

It is not needed. `execution_model` is used to navigate to an actual record, when dealing with use cases that are working with an abstract definition of an aspect. The opposite direction is kind of pointless, since the user is already aware to the fact that he is dealing with an aspect.

> Also, I can't seem to find the table definition for this?

Tables for inherited classes (actual aspects) will be defined by the extensions that are adding some aspects to host. For example puppet extension will add a `PuppetAspect` model that will inherit from this class, and add a proper migration file.